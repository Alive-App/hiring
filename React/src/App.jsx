import React from 'react';

import './App.scss';
import { Provider } from 'react-redux';
import store from './config/store';
import Navigation from './pages';
import { Toast } from 'primereact/toast';
import { toasterRef } from './utils/toaster';

function App() {
  return (
    <div className="App">
      <Provider store={store}>
        <Navigation />
      </Provider>
      <Toast ref={toasterRef} />
    </div>
  );
}

export default App;
