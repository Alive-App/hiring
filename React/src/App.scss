.App {
  // padding: 10px;
  // text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.header {
  padding: 10px 20px;
  background-color: #eee;
  // display: inline;
  .stock-search {
    display: flex;
    align-items: center;
    justify-content: flex-end;
    align-self: flex-end;
    button {
      margin-left: 10px;
    }
  }
}
.p-dialog-content {
  height: 90vh;
}
