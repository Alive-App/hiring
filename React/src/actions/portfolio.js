import store from '../config/store';
import { ADD_TO_PORTFOLIO } from './types';

const dispatch = store.dispatch;
const getState = store.getState;

export const addToPortfolio = (payload) => {
  dispatch({ type: ADD_TO_PORTFOLIO, payload });
};
export const addFromPortfolio = (payload) => {
  dispatch({ type: ADD_TO_PORTFOLIO, payload });
};
