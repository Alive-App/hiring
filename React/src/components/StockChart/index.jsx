import React from 'react';
import Chart from 'react-google-charts';

import { Skeleton } from 'primereact/skeleton';
import { ProgressBar } from 'primereact/progressbar';
import { TabMenu } from 'primereact/tabmenu';
import { Message } from 'primereact/message';
import { Button } from 'primereact/button';

import { useAPI } from '../../utils/api';
import { parseISO } from 'date-fns';
import { defaultData } from '../StockDetails';
import { shallowEqual, useSelector } from 'react-redux';

const TABS = [
  { value: 'intraday', label: '24h', icon: 'pi pi-fw pi-calendar' },
  { value: 'daily', label: 'Diário', icon: 'pi pi-fw pi-calendar' },
  { value: 'weekly', label: 'Semanal', icon: 'pi pi-fw pi-calendar' },
  { value: 'monthly', label: 'Mensal', icon: 'pi pi-fw pi-calendar' },
];
function StockChart({ id }) {
  const [activeType, setTabItem] = React.useState(TABS[1]);

  const [type, setType] = React.useState('daily');

  const [{ loading: loadingChart, error: chartError }, fetchChart] = useAPI('daily');

  const storeChartData = useSelector(
    (state) => ({
      onPortfolio: state.portfolio.findIndex(({ Symbol }) => Symbol == id) > -1,
      intraday: (state.history.intraday || {})[id] || defaultData,
      daily: (state.history.daily || {})[id] || defaultData,
      weekly: (state.history.weekly || {})[id] || defaultData,
      monthly: (state.history.monthly || {})[id] || defaultData,
    }),
    shallowEqual,
  );

  const refreshChart = React.useCallback(
    async (id) => {
      await fetchChart({ fn: type, params: [id, 'compact', 'json', '15min'] });
    },
    [type],
  );

  const chartList = React.useMemo(
    () =>
      storeChartData[type]
        ? storeChartData[type].list.map((v, i) => {
            if (i && typeof v[0] == 'string') v[0] = parseISO(v[0]);
            return v;
          }) || []
        : [],
    [storeChartData[type]],
  );

  React.useEffect(() => {
    if (id) refreshChart(id);
  }, [refreshChart]);

  return (
    <>
      <TabMenu
        model={TABS}
        activeItem={activeType}
        onTabChange={(e) => {
          setTabItem(e.value);
          setType(e.value.value);
        }}
      />
      <div className="p-card chart-container p-mt-3 p-d-flex p-p-3 p-flex-column">
        {!!loadingChart && <ProgressBar className="top-progress" mode="indeterminate" />}
        {!!chartError && (
          <div className="p-col-12">
            <Message
              severity="warn"
              text={
                <>
                  (Dados offline): {chartError}{' '}
                  <Button label="tentar novamente" className="p-button-text" onClick={() => refreshChart(id)} />
                </>
              }
            />
          </div>
        )}
        {loadingChart && (
          <>
            <Skeleton width="400px" height="25px" />
            <Skeleton className="p-mt-2" width="100%" height="150px" />
          </>
        )}
        {!!chartList.length && !loadingChart && (
          <>
            <Chart
              chartType="LineChart"
              loader={<div className="p-d-flex p-justify-center p-align-center">Desenhando gráficos...</div>}
              data={chartList}
              options={{
                hAxis: { title: 'Tempo' },
                vAxis: { title: 'Valor ' },
                series: { 1: { curveType: 'function' } },
              }}
              rootProps={{ 'data-testid': '2' }}
              controls={
                chartList.length
                  ? [
                      {
                        controlType: 'DateRangeFilter',

                        options: {
                          filterColumnLabel: 'x',

                          ui: { format: { pattern: 'dd-MM-yyyy' } },
                        },
                      },
                    ]
                  : null
              }
            />
          </>
        )}
      </div>
    </>
  );
}

export default StockChart;
