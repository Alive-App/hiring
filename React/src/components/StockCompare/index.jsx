import { Button } from 'primereact/button';
import { Skeleton } from 'primereact/skeleton';
import React from 'react';
import Chart from 'react-google-charts';

import { useSelector } from 'react-redux';

import { SearchContext } from '../../pages';
import { useAPI } from '../../utils/api';
import { defaultData } from '../StockDetails';

// import { Container } from './styles';

function StockCompare({ id }) {
  const [{ loading, error }, fetchData] = useAPI('daily');
  const { open, close } = React.useContext(SearchContext);
  const [stocks, setStocks] = React.useState([]);
  const dailyList = useSelector((state) => state.history.daily);
  const addStock = React.useCallback(async (d) => {
    if (d) {
      const { value: stock } = d;
      setStocks((old) => [stock, ...old]);
      close();
      await fetchData({ params: [stock.symbol, 'compact', 'json', '5min'] });
    }
  }, []);

  const chartData = React.useMemo(() => {
    let v = [['Nome', 'Valor', 'Alta', 'Baixa']];
    const arr = [...stocks];
    arr.unshift({ symbol: id });

    for (let i = 0; i < arr.length; i++) {
      const { symbol } = arr[i];

      const { list = [] } = (dailyList || {})[symbol] || defaultData;
      const [_, close = 0, high = 0, low = 0] = list[1] || [];
      v.push([symbol, close, high, low]);
    }
    return v;
  }, [stocks, id, dailyList]);

  return (
    <div className="p-card p-p-2  p-mt-2">
      <div className="p-grid">
        <div className="p-col p-text-bold p-d-flex p-align-center">
          Comparar ação
          <Button icon="pi pi-plus" label="Adicionar ação" className="p-ml-3 p-button-sm p-button-text" onClick={() => open(addStock)} />
        </div>
        <div className="p-col"></div>
      </div>
      <div>
        {stocks.map(({ symbol, name }, i) => {
          const diff = (chartData[i + 2][1] - chartData[1][1]).toFixed(2);

          return (
            <div key={i} className="p-d-flex p-flex-row p-justify-between p-text-secondary">
              <div className="p-col">
                <span className="p-text-bold" style={{ fontSize: 20 }}>
                  {symbol}
                </span>{' '}
                {name}{' '}
                <span className="p-text-bold">
                  {chartData[i + 2][1]}
                  <span style={{ color: diff < 0 ? 'red' : 'green' }}> ({diff})</span>
                </span>
              </div>
              <div className="p-col-2">
                <Button
                  icon="pi pi-minus"
                  className="p-button-danger"
                  onClick={() =>
                    setStocks((old) => {
                      old.splice(i, 1);
                      return [...old];
                    })
                  }
                />
              </div>
            </div>
          );
        })}
        {loading && <Skeleton width="85vw" height="400px" />}
        {!loading && !!stocks.length && (
          <Chart
            height="400px"
            chartType="BarChart"
            loader={<div>Desenhando gráficos</div>}
            data={chartData}
            options={{
              title: '',
              chartArea: { width: '50%' },
              hAxis: {
                title: 'Valor',
                minValue: 0,
              },
            }}
            rootProps={{ 'data-testid': '1' }}
          />
        )}
      </div>
    </div>
  );
}

export default StockCompare;
