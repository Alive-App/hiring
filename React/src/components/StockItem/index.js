import React from 'react';

// import { Container } from './styles';

const StockItem = ({ data, onClick }) => {
  return (
    <div className="p-card p-mt-2" style={{ cursor: 'pointer' }} onClick={onClick}>
      <div className="product-item">
        <div className="product-detail">
          <div className="p-d-flex p-flex-row p-justify-between">
            <div className="product-name">{data.name || data.Name}</div>
            <div>
              <i className="pi pi-flag product-category-icon"></i>
              <span className="product-category">
                {data.region || data.Country} ({data.currency || data.Currency})
              </span>
            </div>
          </div>
          <div className="product-description">{data.symbol || data.Symbol}</div>
        </div>
      </div>
    </div>
  );
};

export default StockItem;
