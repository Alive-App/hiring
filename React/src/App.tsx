import React from 'react';
import logo from './logo.svg';
import './App.css';
import { Provider } from 'react-redux';
import store from './config/store';
function App() {
	return (
		<div className='App'>
			<Provider store={store}>
				<h1>ok</h1>
			</Provider>
		</div>
	);
}

export default App;
