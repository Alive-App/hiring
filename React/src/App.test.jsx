import React from 'react';

import { mount } from 'enzyme';
import App from './App';
import toJson from 'enzyme-to-json';

import StockItem from './components/StockItem';
import store from './config/store';
import { Provider } from 'react-redux';
import Stock from './pages/Stock';
import StockDetails from './components/StockDetails';
import StockChart from './components/StockChart';
import StockCompare from './components/StockCompare';
import StockProjection from './components/StockProjection';
import Home from './pages/Home';
import { addToPortfolio, removeFromPortfolio } from './actions/portfolio';

describe('rendering components', () => {

  it('should render app without crashing', () => {
    const app = mount(<App />);
    expect(toJson(app)).toMatchSnapshot();
  });

  it('should render header', () => {
    const app = mount(<App />);
    expect(app.find('#header')).toHaveLength(1);
  });

  it('should open search modal', () => {
    const app = mount(<App />);

    app.find('#search-btn').forEach((el) => el.simulate('click'));

    expect(app.find('#search-input')).toHaveLength(1);
  });

  it('should type some search', () => {
    const app = mount(<App />);
    app.find('#search-btn').forEach((el) => el.simulate('click'));
    const input = app.find('#search-input');
    input.simulate('focus');
    input.instance().value = 'apple';

    expect(input.instance().value).toBe('apple');
  });

  it('should render a stock item', () => {
    const item = mount(<StockItem data={{ symbol: 'AAPL', name: 'Apple inc' }} />);
    expect(toJson(item.find('.product-name'))).toMatchSnapshot();
  });

  it('should render the Stock screen without crashing', () => {
    const screen = mount(
      <Provider store={store}>
        <Stock id="AAPL" />
        );
      </Provider>,
    );
    expect(screen.find('StockDetails')).toHaveLength(1);
  });

  it('should render StockDetails without crashing', () => {
    const screen = mount(
      <Provider store={store}>
        <StockDetails id="AAPL" />
        );
      </Provider>,
    );
    expect(toJson(screen)).toMatchSnapshot();
  });

  it('should render StockChart without crashing', () => {
    const screen = mount(
      <Provider store={store}>
        <StockChart id="AAPL" />
        );
      </Provider>,
    );
    expect(toJson(screen)).toMatchSnapshot();
  });

  it('should render StockCompare without crashing', () => {
    const screen = mount(
      <Provider store={store}>
        <StockCompare id="AAPL" />
        );
      </Provider>,
    );
    expect(toJson(screen)).toMatchSnapshot();
  });

  it('should render StockProjection without crashing', () => {
    const screen = mount(
      <Provider store={store}>
        <StockProjection id="AAPL" />
        );
      </Provider>,
    );
    expect(screen.contains(<h5>Projeção de ganhos</h5>)).toBe(true);
  });

  it('should render StockCompare without crashing', () => {
    const screen = mount(
      <Provider store={store}>
        <StockCompare id="AAPL" />
        );
      </Provider>,
    );
    expect(toJson(screen)).toMatchSnapshot();
  });

  it('should render Home portfolio items without crashing', () => {
    addToPortfolio({
      Symbol: 'GOOG',
      Name: 'Alphabet Inc',
      Currency: 'USD',
      Country: 'USA',
    });

    const screen = mount(
      <Provider store={store}>
        <Home id="AAPL" />
        );
      </Provider>,
    );
    screen.find('StockItem');
    expect(screen).toHaveLength(1);
  });
});

describe('test actions', () => {
  beforeAll(() => {
    addToPortfolio({
      Symbol: 'AAPL',
      Name: 'Apple Inc',
      Currency: 'USD',
      Country: 'USA',
    });
  });

  it('should remove a portfolio item', () => {
    console.log('length', store.getState().portfolio.length);
    removeFromPortfolio({
      Symbol: 'AAPL',
    });
    expect(store.getState().portfolio).toHaveLength(1);
  });
});
