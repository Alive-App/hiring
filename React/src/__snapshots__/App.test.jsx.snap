// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering components should render StockChart without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <StockChart
    id="AAPL"
  >
    <TabMenu
      activeItem={
        Object {
          "icon": "pi pi-fw pi-calendar",
          "label": "Diário",
          "value": "daily",
        }
      }
      className={null}
      id={null}
      model={
        Array [
          Object {
            "icon": "pi pi-fw pi-calendar",
            "label": "24h",
            "value": "intraday",
          },
          Object {
            "icon": "pi pi-fw pi-calendar",
            "label": "Diário",
            "value": "daily",
          },
          Object {
            "icon": "pi pi-fw pi-calendar",
            "label": "Semanal",
            "value": "weekly",
          },
          Object {
            "icon": "pi pi-fw pi-calendar",
            "label": "Mensal",
            "value": "monthly",
          },
        ]
      }
      onTabChange={[Function]}
      style={null}
    >
      <div
        className="p-tabmenu p-component"
        id={null}
        style={null}
      >
        <ul
          className="p-tabmenu-nav p-reset"
          role="tablist"
        >
          <li
            aria-expanded={false}
            aria-selected={false}
            className="p-tabmenuitem"
            key="24h_0"
            role="tab"
          >
            <a
              className="p-menuitem-link"
              href="#"
              onClick={[Function]}
              role="presentation"
            >
              <span
                className="p-menuitem-icon pi pi-fw pi-calendar"
              />
              <span
                className="p-menuitem-text"
              >
                24h
              </span>
              <Ripple />
            </a>
          </li>
          <li
            aria-expanded={true}
            aria-selected={true}
            className="p-tabmenuitem p-highlight"
            key="Diário_1"
            role="tab"
          >
            <a
              className="p-menuitem-link"
              href="#"
              onClick={[Function]}
              role="presentation"
            >
              <span
                className="p-menuitem-icon pi pi-fw pi-calendar"
              />
              <span
                className="p-menuitem-text"
              >
                Diário
              </span>
              <Ripple />
            </a>
          </li>
          <li
            aria-expanded={false}
            aria-selected={false}
            className="p-tabmenuitem"
            key="Semanal_2"
            role="tab"
          >
            <a
              className="p-menuitem-link"
              href="#"
              onClick={[Function]}
              role="presentation"
            >
              <span
                className="p-menuitem-icon pi pi-fw pi-calendar"
              />
              <span
                className="p-menuitem-text"
              >
                Semanal
              </span>
              <Ripple />
            </a>
          </li>
          <li
            aria-expanded={false}
            aria-selected={false}
            className="p-tabmenuitem"
            key="Mensal_3"
            role="tab"
          >
            <a
              className="p-menuitem-link"
              href="#"
              onClick={[Function]}
              role="presentation"
            >
              <span
                className="p-menuitem-icon pi pi-fw pi-calendar"
              />
              <span
                className="p-menuitem-text"
              >
                Mensal
              </span>
              <Ripple />
            </a>
          </li>
          <li
            className="p-tabmenu-ink-bar"
          />
        </ul>
      </div>
    </TabMenu>
    <div
      className="p-card chart-container p-mt-3 p-d-flex p-p-3 p-flex-column"
    >
      <ProgressBar
        className="top-progress"
        color={null}
        displayValueTemplate={null}
        id={null}
        mode="indeterminate"
        showValue={true}
        style={null}
        unit="%"
        value={null}
      >
        <div
          className="p-progressbar p-component p-progressbar-indeterminate top-progress"
          id={null}
          role="progressbar"
          style={null}
        >
          <div
            className="p-progressbar-indeterminate-container"
          >
            <div
              className="p-progressbar-value p-progressbar-value-animate"
              style={
                Object {
                  "backgroundColor": null,
                }
              }
            />
          </div>
        </div>
      </ProgressBar>
      <Skeleton
        animation="wave"
        borderRadius={null}
        className={null}
        height="25px"
        shape="rectangle"
        size={null}
        style={null}
        width="400px"
      >
        <div
          className="p-skeleton p-component"
          style={
            Object {
              "borderRadius": null,
              "height": "25px",
              "width": "400px",
            }
          }
        />
      </Skeleton>
      <Skeleton
        animation="wave"
        borderRadius={null}
        className="p-mt-2"
        height="150px"
        shape="rectangle"
        size={null}
        style={null}
        width="100%"
      >
        <div
          className="p-skeleton p-component p-mt-2"
          style={
            Object {
              "borderRadius": null,
              "height": "150px",
              "width": "100%",
            }
          }
        />
      </Skeleton>
    </div>
  </StockChart>
  );
</Provider>
`;

exports[`rendering components should render StockCompare without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <StockCompare
    id="AAPL"
  >
    <div
      className="p-card p-p-2  p-mt-2"
    >
      <div
        className="p-grid"
      >
        <div
          className="p-col p-text-bold p-d-flex p-align-center"
        >
          Comparar ação
          <Button
            badge={null}
            badgeClassName={null}
            className="p-ml-3 p-button-sm p-button-text"
            icon="pi pi-plus"
            iconPos="left"
            label="Adicionar ação"
            onClick={[Function]}
            tooltip={null}
            tooltipOptions={null}
          >
            <button
              className="p-button p-component p-ml-3 p-button-sm p-button-text"
              onClick={[Function]}
            >
              <span
                className="pi pi-plus p-c p-button-icon-left"
              />
              <span
                className="p-button-label p-c"
              >
                Adicionar ação
              </span>
              <Ripple />
            </button>
          </Button>
        </div>
        <div
          className="p-col"
        />
      </div>
      <div />
    </div>
  </StockCompare>
  );
</Provider>
`;

exports[`rendering components should render StockCompare without crashing 2`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <StockCompare
    id="AAPL"
  >
    <div
      className="p-card p-p-2  p-mt-2"
    >
      <div
        className="p-grid"
      >
        <div
          className="p-col p-text-bold p-d-flex p-align-center"
        >
          Comparar ação
          <Button
            badge={null}
            badgeClassName={null}
            className="p-ml-3 p-button-sm p-button-text"
            icon="pi pi-plus"
            iconPos="left"
            label="Adicionar ação"
            onClick={[Function]}
            tooltip={null}
            tooltipOptions={null}
          >
            <button
              className="p-button p-component p-ml-3 p-button-sm p-button-text"
              onClick={[Function]}
            >
              <span
                className="pi pi-plus p-c p-button-icon-left"
              />
              <span
                className="p-button-label p-c"
              >
                Adicionar ação
              </span>
              <Ripple />
            </button>
          </Button>
        </div>
        <div
          className="p-col"
        />
      </div>
      <div />
    </div>
  </StockCompare>
  );
</Provider>
`;

exports[`rendering components should render StockDetails without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <StockDetails
    id="AAPL"
  >
    <div
      className="p-card p-mb-2 p-p-2"
    >
      <ProgressBar
        className="top-progress"
        color={null}
        displayValueTemplate={null}
        id={null}
        mode="indeterminate"
        showValue={true}
        style={null}
        unit="%"
        value={null}
      >
        <div
          className="p-progressbar p-component p-progressbar-indeterminate top-progress"
          id={null}
          role="progressbar"
          style={null}
        >
          <div
            className="p-progressbar-indeterminate-container"
          >
            <div
              className="p-progressbar-value p-progressbar-value-animate"
              style={
                Object {
                  "backgroundColor": null,
                }
              }
            />
          </div>
        </div>
      </ProgressBar>
      <div
        className="p-grid"
      >
        <div
          className="p-col"
        >
          <h3
            className="p-m-0 p-d-flex p-align-center"
            style={
              Object {
                "fontSize": "18pt",
              }
            }
          >
            <Button
              badge={null}
              badgeClassName={null}
              className="p-button-text"
              disabled={true}
              icon="pi pi-star-o"
              iconPos="left"
              label={null}
              onClick={[Function]}
              tooltip={null}
              tooltipOptions={null}
            >
              <button
                className="p-button p-component p-button-text p-button-icon-only p-disabled"
                disabled={true}
                onClick={[Function]}
              >
                <span
                  className="pi pi-star-o p-c"
                />
                <span
                  className="p-button-label p-c"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&nbsp;",
                    }
                  }
                />
                <Ripple />
              </button>
            </Button>
          </h3>
          <div
            className="p-d-flex p-flex-row p-justify-between p-text-secondary"
          >
            <div>
              <span
                className="p-mr-2"
                id="symbol"
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              />
              <span
                className="p-text-bold"
                id="close"
                style={
                  Object {
                    "fontSize": 16,
                  }
                }
              >
                 
                0.00
              </span>
               
              <span
                id="diff"
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                 0.00 (0.00%)
              </span>
              <span
                className="p-ml-2"
                id="high"
              >
                <i
                  className="pi pi-arrow-up"
                  style={
                    Object {
                      "color": "green",
                    }
                  }
                />
                 
                <span>
                  0.00
                </span>
              </span>
              <span
                className="p-ml-2"
                id="low"
              >
                <i
                  className="pi pi-arrow-down"
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                />
                 
                <span>
                  0.00
                </span>
              </span>
            </div>
            <span />
          </div>
        </div>
      </div>
      <Tooltip
        appendTo={null}
        at={null}
        autoZIndex={true}
        baseZIndex={0}
        className="p-ml-2"
        content="Ação"
        event={null}
        hideDelay={0}
        hideEvent="mouseleave"
        mouseTrack={false}
        mouseTrackLeft={5}
        mouseTrackTop={5}
        my={null}
        onBeforeHide={null}
        onBeforeShow={null}
        onHide={null}
        onShow={null}
        position="top"
        showDelay={0}
        showEvent="mouseenter"
        style={null}
        target="#symbol"
        updateDelay={0}
      />
      <Tooltip
        appendTo={null}
        at={null}
        autoZIndex={true}
        baseZIndex={0}
        className="p-ml-2"
        content="Valor"
        event={null}
        hideDelay={0}
        hideEvent="mouseleave"
        mouseTrack={false}
        mouseTrackLeft={5}
        mouseTrackTop={5}
        my={null}
        onBeforeHide={null}
        onBeforeShow={null}
        onHide={null}
        onShow={null}
        position="top"
        showDelay={0}
        showEvent="mouseenter"
        style={null}
        target="#close"
        updateDelay={0}
      />
      <Tooltip
        appendTo={null}
        at={null}
        autoZIndex={true}
        baseZIndex={0}
        className="p-ml-2"
        content="Diferença e percentual em relação ao dia anterior"
        event={null}
        hideDelay={0}
        hideEvent="mouseleave"
        mouseTrack={false}
        mouseTrackLeft={5}
        mouseTrackTop={5}
        my={null}
        onBeforeHide={null}
        onBeforeShow={null}
        onHide={null}
        onShow={null}
        position="top"
        showDelay={0}
        showEvent="mouseenter"
        style={null}
        target="#diff"
        updateDelay={0}
      />
      <Tooltip
        appendTo={null}
        at={null}
        autoZIndex={true}
        baseZIndex={0}
        className="p-ml-2"
        content="Alta"
        event={null}
        hideDelay={0}
        hideEvent="mouseleave"
        mouseTrack={false}
        mouseTrackLeft={5}
        mouseTrackTop={5}
        my={null}
        onBeforeHide={null}
        onBeforeShow={null}
        onHide={null}
        onShow={null}
        position="top"
        showDelay={0}
        showEvent="mouseenter"
        style={null}
        target="#high"
        updateDelay={0}
      />
      <Tooltip
        appendTo={null}
        at={null}
        autoZIndex={true}
        baseZIndex={0}
        className="p-ml-2"
        content="Baixa"
        event={null}
        hideDelay={0}
        hideEvent="mouseleave"
        mouseTrack={false}
        mouseTrackLeft={5}
        mouseTrackTop={5}
        my={null}
        onBeforeHide={null}
        onBeforeShow={null}
        onHide={null}
        onShow={null}
        position="top"
        showDelay={0}
        showEvent="mouseenter"
        style={null}
        target="#low"
        updateDelay={0}
      />
    </div>
  </StockDetails>
  );
</Provider>
`;

exports[`rendering components should render StockProjection without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <StockProjection
    id="AAPL"
  >
    <div
      className="p-card p-p-2 p-mt-2"
    >
      <h5>
        Projeção de ganhos
      </h5>
      <div
        className="p-grid"
      >
        <div
          className="p-col p-col-2"
        >
          <input
            className="p-inputtext"
            defaultValue="1"
            name="qty"
            placeholder="quantidade"
            required={true}
          />
        </div>
        <div
          className="p-col p-col-2"
        >
          <input
            className="p-inputtext"
            name="start_date"
            required={true}
            type="date"
          />
        </div>
        <div
          className="p-col p-col-2"
        >
          <input
            className="p-inputtext"
            max={2021-01-25T02:58:42.424Z}
            name="end_date"
            required={true}
            type="date"
          />
        </div>
        <div
          className="p-col p-col-1"
        >
          <Button
            badge={null}
            badgeClassName={null}
            icon="pi pi-search"
            iconPos="left"
            label={null}
            tooltip={null}
            tooltipOptions={null}
          >
            <button
              className="p-button p-component p-button-icon-only"
            >
              <span
                className="pi pi-search p-c"
              />
              <span
                className="p-button-label p-c"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&nbsp;",
                  }
                }
              />
              <Ripple />
            </button>
          </Button>
        </div>
        <div
          className="p-col p-col-3"
        >
          Lucro: 
          <span
            className="p-text-bold "
            style={
              Object {
                "color": "green",
                "fontSize": 20,
              }
            }
          >
            0.00
          </span>
        </div>
      </div>
    </div>
  </StockProjection>
  );
</Provider>
`;

exports[`rendering components should render a stock item 1`] = `
<div
  className="product-name"
>
  Apple inc
</div>
`;

exports[`rendering components should render app without crashing 1`] = `
<App>
  <div
    className="App"
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Navigation>
        <Header>
          <header
            className="header p-flex-row p-d-flex p-justify-between"
            id="header"
          >
            <div />
            <Button
              badge={null}
              badgeClassName={null}
              className="p-button-outlined"
              icon="pi pi-search"
              iconPos="left"
              id="search-btn"
              label="Buscar ações"
              onClick={[Function]}
              tooltip={null}
              tooltipOptions={null}
            >
              <button
                className="p-button p-component p-button-outlined"
                id="search-btn"
                onClick={[Function]}
              >
                <span
                  className="pi pi-search p-c p-button-icon-left"
                />
                <span
                  className="p-button-label p-c"
                >
                  Buscar ações
                </span>
                <Ripple />
              </button>
            </Button>
          </header>
        </Header>
        <SearchDialog
          onSearch={null}
        >
          <Dialog
            appendTo={null}
            ariaCloseIconLabel="Close"
            baseZIndex={0}
            blockScroll={false}
            className={null}
            closable={true}
            closeOnEscape={true}
            contentClassName={null}
            contentStyle={null}
            dismissableMask={false}
            focusOnShow={true}
            footer={null}
            header="Pesquisar"
            icons={null}
            id="search-modal"
            maskClassName={null}
            maximizable={false}
            maximized={false}
            modal={true}
            onHide={[Function]}
            onMaximize={null}
            onShow={null}
            position="center"
            rtl={false}
            showHeader={true}
            style={
              Object {
                "height": "90vh",
                "maxWidth": 768,
              }
            }
            visible={false}
          />
        </SearchDialog>
        <div
          className="p-p-4"
        >
          <Home>
            <div
              className="p-mt-4"
            >
              <h3>
                Meu portfolio
              </h3>
              <div
                className="p-col-4 p-col"
              >
                <h5>
                  Lista vazia, busque uma empresa e favorite-a para aparecer aqui.
                </h5>
              </div>
            </div>
          </Home>
        </div>
      </Navigation>
    </Provider>
    <Toast
      baseZIndex={0}
      className={null}
      id={null}
      onClick={null}
      onRemove={null}
      position="top-right"
      style={null}
    >
      <div
        className="p-toast p-component p-toast-top-right"
        id={null}
        style={null}
      >
        <TransitionGroup
          childFactory={[Function]}
          component="div"
        >
          <div />
        </TransitionGroup>
      </div>
    </Toast>
  </div>
</App>
`;
